src-tauri/src/platform_specific/windows.rs

```diff
@@ -1,34 +1,34 @@
 /// Windows specific code
 use anyhow::{anyhow, Result};
 
-use crate::{check_is_valid_game_path, constants::TITANFALL2_ORIGIN_IDS};
+#[cfg(target_os = "windows")]
+use winreg::{enums::HKEY_LOCAL_MACHINE, RegKey};
+
+use crate::check_is_valid_game_path;
 
 /// Gets Titanfall2 install location on Origin
 pub fn origin_install_location_detection() -> Result<String, anyhow::Error> {
-    // Iterate over known Titanfall2 Origin IDs
-    for origin_id in TITANFALL2_ORIGIN_IDS {
-        match game_scanner::origin::find(origin_id) {
-            // Origin ID found as installed game
-            Ok(game) => {
-                if game.path.is_some() {
-                    let game_path = game.path.unwrap();
-                    let game_path_str = game_path.to_str().unwrap();
-                    match check_is_valid_game_path(game_path_str) {
-                        Ok(()) => {
-                            return Ok(game_path_str.to_string());
-                        }
-                        Err(err) => {
-                            log::warn!("{}", err);
-                            continue; // Not a valid game path
-                        }
+    #[cfg(target_os = "windows")]
+    {
+        let hklm = RegKey::predef(HKEY_LOCAL_MACHINE);
+        match hklm.open_subkey("SOFTWARE\\Respawn\\Titanfall2") {
+            Ok(tf) => {
+                let game_path_str: String = tf.get_value("Install Dir")?;
```

Comment:
the type that `get_value` returns depend on the type that is expected, so we cannot easily set the type of game_path_str to something like `Result<String>` without the function returning a `Result<Result<String>>`.

The curent way its done is a bit ugly but it does the job (plus what are the chances someone has the registry subkey but not the value :clueless:)

Reply:
I can probably figure out a way to improve this but given how many dependencies we can slash by removing game scanner I'm fine with merging as is ^^